src/main/java/com/onlinebookstore/onlinebookstore/controller/AuthenticationController.java

```diff
@@ -0,0 +1,33 @@
+package com.onlinebookstore.onlinebookstore.controller;
+
+import com.onlinebookstore.onlinebookstore.dto.user.UserRegistrationRequestDto;
+import com.onlinebookstore.onlinebookstore.dto.user.UserResponseDto;
+import com.onlinebookstore.onlinebookstore.exeption.RegistrationException;
+import com.onlinebookstore.onlinebookstore.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "User management", description = "Endpoints for managing users")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "api/auth")
+public class AuthenticationController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
+    @Operation(summary = "Create new User",
+            description = "Email, password and repeatpassword cannot be empty")

```

Comment:
this description describes body and not the endpoint or operation itself, it should be part of the `UserRegistrationRequestDto` class then